### PR TITLE
Refactor code upload challenge worker for submission stuck

### PR DESCRIPTION
**Pull request by savish28**
Context: In the Code upload challenge, if the job is not present on the cluster because of any external means (deleted etc). The submission message is stuck in the queue.

- [x] Refactor code upload challenge worker for submission stuck

This PR is in relation to #3452, would update this PR or 3452 based on whichever is merged first.

@KhalidRmb 
cc @Ram81

### CODE REVIEW

File: scripts/workers/code_upload_submission_worker.py [hunk 1]
```diff
@@ -276,47 +276,62 @@ def update_failed_jobs_and_send_logs(
...
+            "submission": submission_pk,
+            "stdout": "",
+            "stderr": "Submission Job failed.",
+            "submission_status": "failed",
+            "result": "[]",
+            "metadata": "",
+        }
+        response = evalai.update_submission_data(
+            submission_data, challenge_pk, phase_pk
+        )
```
Comment (Ram81) [inline scripts/workers/code_upload_submission_worker.py:334]: @savish28 the submission message also needs to be deleted if the job was deleted on the cluster.

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: scripts/workers/code_upload_submission_worker.py (hunk 1 above)
Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:335]: @Ram81 Sure, Done.

File: scripts/workers/code_upload_submission_worker.py [hunk 2]
```diff
@@ -469,48 +469,72 @@ def update_failed_jobs_and_send_logs(
...
+        message_receipt_handle = message.get("receipt_handle")
+        try:
+            delete_job(api_instance, job_name)
+            evalai.delete_message_from_sqs_queue(message_receipt_handle)
+        except Exception as e:
+            logger.exception("Failed to delete submission job: {}".format(e))
+            # Delete message from sqs queue to avoid re-triggering job delete
+            evalai.delete_message_from_sqs_queue(message_receipt_handle)
```
Comment (Ram81) [inline scripts/workers/code_upload_submission_worker.py:537]: @savish28 can we move this logic of marking submission failed, deleting job and deleting message to a separate method named `cleanup_submission` where you pass the required details. The current code is not readable.

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: scripts/workers/code_upload_submission_worker.py (hunk 2 above)
Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:537]: Yeah, makes sense.

Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py]: @Ram81 Updated the code, Do we want something like this.

**Comment by savish28** [comment]
@Ram81 ping.

File: scripts/workers/code_upload_submission_worker.py [hunk 3]
```diff
@@ -461,6 +461,46 @@ def read_job(api_instance, job_name):
...
+    job_name,
+    submission_pk,
+    challenge_pk,
+    phase_pk,
+    stderr,
+    message,
+):
+    """Function to update status of submission to EvalAi, Delete corrosponding job from cluster and messaage from SQS.
+    Arguments:
+        api_instance {[AWS EKS API object]} -- API object for deleting job
```
Comment (Ram81) [inline scripts/workers/code_upload_submission_worker.py:476]: @savish28 please add description for all params if adding docstrings

**Review by Ram81 — CHANGES_REQUESTED** [review]
Minor change. Rest of it LGTM

File: scripts/workers/code_upload_submission_worker.py (hunk 3 above)
Comment (savish28) [reply scripts/workers/code_upload_submission_worker.py:476]: @Ram81 Updated.

**Review by Ram81 — APPROVED** [review]
LGTM 👍

**Pull request merged**